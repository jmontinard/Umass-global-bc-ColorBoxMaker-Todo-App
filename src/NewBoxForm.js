import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';

function NewBoxForm({ createBox }) {
  const [formData, setFormData] = useState({
    width: '',
    height: '',
    backgroundColor: '',
  });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const newBox = { ...formData, id: uuidv4() };
    createBox(newBox);
    setFormData({
      width: '',
      height: '',
      backgroundColor: '',
    });
  };

  return (
    <form onSubmit={handleSubmit}>
      <div>
        <label htmlFor="width">Width:</label>
        <input
          type="text"
          id="width"
          name="width"
          value={formData.width}
          onChange={handleChange}
        />
      </div>
      <div>
        <label htmlFor="height">Height:</label>
        <input
          type="text"
          id="height"
          name="height"
          value={formData.height}
          onChange={handleChange}
        />
      </div>
      <div>
        <label htmlFor="backgroundColor">Background Color:</label>
        <input
          type="text"
          id="backgroundColor"
          name="backgroundColor"
          value={formData.backgroundColor}
          onChange={handleChange}
        />
      </div>
      <button type="submit">Add Box</button>
    </form>
  );
}

export default NewBoxForm;
