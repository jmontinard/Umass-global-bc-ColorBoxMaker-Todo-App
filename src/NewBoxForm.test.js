import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';  
import NewBoxForm from './NewBoxForm';

// test('NewBoxForm component renders without crashing', () => {
//   render(<NewBoxForm createBox={() => {}} />);
// });

// test('NewBoxForm component matches snapshot', () => {
//   const { asFragment } = render(<NewBoxForm createBox={() => {}} />);
//   expect(asFragment()).toMatchSnapshot();
// });

// test('NewBoxForm allows the user to enter data and submit a new box', () => {
//   const createBox = jest.fn();
//   const { getByLabelText, getByRole } = render(
//     <NewBoxForm createBox={createBox} />
//   );

//   // Simulate user typing into the form inputs
//   userEvent.type(getByLabelText(/width:/i), '100');
//   userEvent.type(getByLabelText(/height:/i), '150');
//   userEvent.type(getByLabelText(/background color:/i), 'blue');

//   // Simulate form submission
//   fireEvent.submit(getByRole('button'));

//   // Check if the createBox function was called once
//   expect(createBox).toHaveBeenCalledTimes(1);

//   // Check if the createBox function was called with the correct box data
//   expect(createBox).toHaveBeenCalledWith(
//     expect.objectContaining({
//       width: '100',
//       height: '150',
//       backgroundColor: 'blue',
//     })
//   );
// });

test('renders NewBoxForm component', () => {
  const { container } = render(<NewBoxForm />);
  expect(container).toBeInTheDocument();
});

test('matches snapshot for NewBoxForm component', () => {
  const { container } = render(<NewBoxForm />);
  expect(container).toMatchSnapshot();
});

test('updates form data on input change', () => {
  const { getByLabelText } = render(<NewBoxForm />);

  act(() => {
    fireEvent.change(getByLabelText('Color'), { target: { value: 'red' } });
    fireEvent.change(getByLabelText('Width'), { target: { value: '20' } });
    fireEvent.change(getByLabelText('Height'), { target: { value: '30' } });
  });

  expect(getByLabelText('Color').value).toBe('red');
  expect(getByLabelText('Width').value).toBe('20');
  expect(getByLabelText('Height').value).toBe('30');
});

