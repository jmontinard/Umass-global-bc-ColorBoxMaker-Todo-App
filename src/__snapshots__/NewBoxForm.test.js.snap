// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBoxForm component matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="backgroundColor"
      >
        Background Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        type="text"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Add Box
    </button>
  </form>
</DocumentFragment>
`;

exports[`matches snapshot for NewBoxForm component 1`] = `
<div>
  <form>
    <div>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="backgroundColor"
      >
        Background Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        type="text"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Add Box
    </button>
  </form>
</div>
`;
